src/nav_msg_glo.c

```diff
@@ -0,0 +1,285 @@
+    word = extract_word_glo(n, bit_index+1, s) |
+           extract_word_glo(n, bit_index+1 + s, n_bits - s) << s;
+  }
+
+  return word;
+}
+
+/** Navigation message GLO decoding update.
+ * Called once per nav bit interval (10 ms).
+ * Performs the necessary steps to store the nav bits in buffer.
+ *
+ * \param n GLO Nav message decode state struct
+ * \param bit_val State of the nav bit to process, 0 or 1
+ *
+ * \return  1 if Glo nav string ready for decoding,
+ *         -1 otherwise.
+ */
+s8 nav_msg_update_glo(nav_msg_glo_t *n, bool bit_val)
+{
+  s8 ret = -1;
+
+  switch (n->state) {
+    case SYNC_TM: /* try to find time mark */
+      /* put incoming bit at the tail of the buffer */
+      n->string_bits[0] <<= 1; /* use one word of buffer for that purpose */
+      n->string_bits[0] |= bit_val;
+      /* collected bits match time mark? if not stay at this state */
+      if(extract_word_glo(n, 1, GLO_TM_LEN) == GLO_TM) {
+        /* time mark found, next time start collecting data bits */
+        n->meander_bits_cnt = 0;
+        n->manchester = 0;
+        n->state = GET_DATA_BIT;
+        n->string_bits[0] = 0;
+      }
+      break;
+    case GET_DATA_BIT: /* collect data bits of string */
+      n->meander_bits_cnt++;
+      n->manchester <<= 1;
+      n->manchester |= bit_val; /* store incoming bit */
+      /* did we take 2 bits of line code?
+       * if yes, remove meander and store bit in buffer,
+       * if no, stay at the state */
+      if (n->meander_bits_cnt == 2) {
+        /* shift whole buffer by 1 bit left */
+        for (u8 i = NAV_MSG_GLO_STRING_BITS_LEN - 1; i > 0; i--) {
+          u32 tmp = (n->string_bits[i] << 1) | ((n->string_bits[i-1] & (1<<31)) >> 31);
+          n->string_bits[i] = tmp;
+        }
+        n->string_bits[0] <<= 1;
+        /* store bit after meander removal to buffer */
+        n->string_bits[0] |= (n->manchester ^ 2) & 1;
+        n->current_head_bit_index++;
+        n->meander_bits_cnt = 0;
+        n->manchester = 0;
+        /* did we received all bits of a string?
+         * if yes, notify user and start searching time mark again*/
+        if (n->current_head_bit_index == 85) {
+          n->current_head_bit_index = 0;
+          n->state = SYNC_TM;
+          ret = 1;
+        }
+      }
+      break;
+    default:
+      nav_msg_init_glo(n); //TODO: probably not needed to initialize next_string_id
+      break;
+  }
+  return ret;
+}
+
+/** The function decodes a GLO navigation string
+ * \param n Pointer to nav_msg_glo_t structure which contains GLO string
+ * \param e Pointer to Ephemeris to store
+ * \return 0 - decode not completed,
+ *         1 -- decode completed, all ephemeris data stored
+ *         <0 -- in case of an error.
+ */
+s8 process_string_glo(nav_msg_glo_t *n, ephemeris_t *e)
+{
+  /* Extract and check dummy bit from GLO string, bit 85 in GLO string */
+  if (extract_word_glo(n, 85, 1) != 0) {
+    log_error("GLO dummy bit is not 0.");
+    return -1;
```

Comment:
No need for 'else' part here. Then the whole 'else' part can be shifted 2 spaces left.